packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -398,13 +421,35 @@ class _CupertinoSearchTextFieldState extends State<CupertinoSearchTextField>
     }
   }
 
+  void _onScroll() {
+    if (_ancestorScrollPosition == null || !_ancestorScrollPosition!.hasPixels) {
```

Comment:
Today we talked about setting `_maxHeight` here when `_onScroll` is called for the first time by looking up the render object as you are below. 👍

Reply:
In that case, we may want to use the ScrollNotificationObserver. It would be more consistent with how the AppBar scrolledUnder behavior works, and we could get more information about what kind of scrolling is occurring.